test/integration/targets/setup_mysql_db/handlers/main.yml

```diff
@@ -0,0 +1,18 @@
+- name: stop mysql service
+  service:
+    name: "{{ mysql_service }}"
+    state: stopped
+  listen: cleanup mysql
+
+- name: remove mysql packages
+  action: '{{ ansible_pkg_mgr }}'
+  args:
+    name: "{{ item }}"
+    state: absent
+  loop: "{{ mysql_packages | union(mysql_cleanup_packages) }}"
+  listen: cleanup mysql
+
+- name: remove pip packages
+  pip:
+    name: mysql-python
```

Comment:
@samdoran , it is because the next (8.0) installation finds files after the previous installation in the default data directory and tries to upgrade them and fails
```Status: \"Data Dictionary upgrade from MySQL 5.7 in progress```

please
a. remove from ```test/integration/targets/setup_mysql_db/handlers/main.yml```
```
- name: remove pip packages
  pip:
    name: mysql-python
    state: absent
```
b. and add to there
```
 15 - name: remove mysql datadir
 16   file:
 17     path: "{{ item }}"
 18     state: absent
 19   loop:
 20   - /var/lib/mysql
 21   - /usr/mysql
 22   listen: cleanup mysql
```
c. also change the block starting from line 9 in ```test/integration/targets/setup_mysql8/tasks/setup_mysql8.yml```
```
  9 - name: Install MySQL community server
 10   yum:
 11     name: '{{ item }}'
 12   loop:
 13   - '{{ mysql_package_name }}'
 14   - 'MySQL-python'
 15   notify: cleanup mysql8
```
it will work, i checked it by running tests you mentioned in your cloned branch

Reply:
> a. remove from test/integration/targets/setup_mysql_db/handlers/main.yml
The `pip` removal is needed to cleanup on FreeBSD.

I added a task to cleanup the data dir. That seems to fix the conflict. 🙌